src/core/encoding.h

```diff
@@ -78,13 +78,13 @@ int encoding_get_utf8_character_bytes(const char input);
  * @param input Input to convert, encoded using UTF-16
  * @param output Output buffer to store the UTF-8 encoded input
  */
-void encoding_utf16_to_utf8(uint16_t *input, uint8_t *output);
+void encoding_utf16_to_utf8(const uint16_t *input, uint8_t *output);
 
 /**
  * Converts an UTF-8 input input to UTF-16 output
  * @param input Input to convert, encoded using UTF-8
  * @param output Output buffer to store the UTF-16 encoded input
  */
-void encoding_utf8_to_utf16(uint8_t *input, uint16_t *output);
+void encoding_utf8_to_utf16(const uint8_t *input, uint16_t *output);
```

Comment:
I know we merged this earlier, but I just realized this signature is confusing. We use `char*` for UTF-8 strings everywhere, and `uint8_t*` for the internally encoded strings. Here we pass UTF-8 in `uint8_t` instead of `char`. This is going to cause weird issues if we mix the two.
Also applies to the previous function.